system_test/aest_sync_SUITE.erl

```diff
@@ -669,10 +671,8 @@ net_split_mining_power(Cfg) ->
     wait_for_value({height, ExtraLength + TargetHeight2}, AllNodes,
                    (ExtraLength + SyncLength) * ?MINING_TIMEOUT, Cfg),
 
-    ct:log("Ping interval set to ~p", [ping_interval()]),
-
     %% Wait at least as long as the ping timer can take
-    try_until(T0 + 2 * ping_interval(),
+    try_until(T0 + 2 * ?PING_INTERVAL,
```

Comment:
Failed at [line 683](https://github.com/aeternity/epoch/pull/1609/commits/8c031688a25475e4a6a08f2620e5f5e4d5b16f65#diff-24519aa66137e7a88004786d3cc88d72R683) on [CI](https://circleci.com/gh/aeternity/epoch/34004):
```
%%% aest_sync_SUITE ==> net_split_recovery: OK
%%% aest_sync_SUITE ==> net_split_mining_power: FAILED
%%% aest_sync_SUITE ==> {{assertEqual,[{module,aest_sync_SUITE},
               {line,683},
               {expression,"B"},
               {expected,#{beneficiary =>
                               <<"ak_2VoAhMd7tVJrDYM5vPJwFRjueZyirDJumVJNeBWL9j1eNTHsRx">>,
                           hash =>
                               <<"kh_sev86KZP4WV1QyJ7QpwaRNFhwM74k3ajszfvHSpqaX7HvD9vr">>,
                           height => 63,
                           miner =>
                               <<"ak_2DeKeakKAdnwrDV2YkiGR1EvTjLN9EeaABCzk3kvNcXpWrhVfr">>,
                           nonce => 11576249076665118194,
                           pow =>
                               [1166,3751,4410,4790,5360,5749,6297,6511,7407,
                                7669,7977,8263,8532,8743,9882,9955,10342,
                                11147,11580,12605,13127,13369,15368,15913,
                                17482,17642,18765,18954,21679,23220,23433,
                                23510,25517,25747,26211,26347,26435,27511,
                                29347,29560,30137,31221],
                           prev_hash =>
                               <<"kh_2Br5KZ6LkW1K9EbF1ksDqyB3WqpBFwmJEXSPuAnnh8dtAVzQpZ">>,
                           prev_key_hash =>
                               <<"kh_2Br5KZ6LkW1K9EbF1ksDqyB3WqpBFwmJEXSPuAnnh8dtAVzQpZ">>,
                           state_hash =>
                               <<"bs_7uoDkytHpPP3q9ubaPMvWQhMLnaHAYwoEsWTJnipn6q87H2WK">>,
                           target => 553694942,time => 1538404780426,
                           version => 24}},
               {value,#{beneficiary =>
                            <<"ak_2VoAhMd7tVJrDYM5vPJwFRjueZyirDJumVJNeBWL9j1eNTHsRx">>,
                        hash =>
                            <<"kh_UHyrLyiTZCej1LEiqxjjDi8GauZraQgPJys1wQgKQnKUyH4gE">>,
                        height => 63,
                        miner =>
                            <<"ak_AdLrEvdcEfGGKbaE69s7PpojWu96L9aaay2FH8aDevnZL55dz">>,
                        nonce => 14242884456010507157,
                        pow =>
                            [986,1168,1247,1672,2787,3182,4619,5345,6756,7664,
                             7666,8803,9280,9828,11400,12267,12802,15893,
                             16356,17148,17477,17645,18545,18931,19488,19834,
                             20039,20163,20462,20582,20619,22339,22399,24446,
                             24489,25191,26189,29891,31270,31580,32016,32470],
                        prev_hash =>
                            <<"kh_2sZaGU1UgGaJoFT3f6DUsUZ6vt6h7d3zimUuoZYm6M8ucRoSXN">>,
                        prev_key_hash =>
                            <<"kh_2sZaGU1UgGaJoFT3f6DUsUZ6vt6h7d3zimUuoZYm6M8ucRoSXN">>,
                        state_hash =>
                            <<"bs_7uoDkytHpPP3q9ubaPMvWQhMLnaHAYwoEsWTJnipn6q87H2WK">>,
                        target => 553713663,time => 1538404824113,
                        version => 24}}]},
 [{aest_sync_SUITE,try_until,2,
                   [{file,"/home/circleci/epoch/_build/system_test+test/extras/system_test/aest_sync_SUITE.erl"},
                    {line,762}]},
  {aest_sync_SUITE,net_split_mining_power,1,
                   [{file,"/home/circleci/epoch/_build/system_test+test/extras/system_test/aest_sync_SUITE.erl"},
                    {line,675}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1539}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1055}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,987}]}]}
```

Reply:
I see in [other CI tests run in master](https://33840-99802036-gh.circle-artifacts.com/0/home/circleci/epoch/system_test/logs/ct_run.epoch_ct%40localhost.2018-09-30_12.03.49/extras.system_test.logs/run.2018-09-30_12.03.49/log_private.22/net1_node1_logs/epoch_sync.log) that `net1_node1` had given up though was contacted after 30s:
```
2018-09-30 12:41:27.308 [debug] <0.1300.0>@aec_peers:connect:673 No more peers available for outbound connections
2018-09-30 12:41:56.094 [debug] <0.6178.0>@aec_peer_connection:accept_init:190 Connection accepted from <<250,77,210,210,140,233,3,34,72,31,42,146,138,127,185,255,107,186,150,211,247,223,38,40,155,204,227,67,84,202,183,9>>
```

Reply:
I see from test case setup that both `net2_node1` and `net2_node3` know `net1_node1`. In a past test run both manage to eventually connect - refs [1](https://33840-99802036-gh.circle-artifacts.com/0/home/circleci/epoch/system_test/logs/ct_run.epoch_ct%40localhost.2018-09-30_12.03.49/extras.system_test.logs/run.2018-09-30_12.03.49/log_private.22/net2_node1_logs/epoch_sync.log) [2](https://33840-99802036-gh.circle-artifacts.com/0/home/circleci/epoch/system_test/logs/ct_run.epoch_ct%40localhost.2018-09-30_12.03.49/extras.system_test.logs/run.2018-09-30_12.03.49/log_private.22/net2_node3_logs/epoch_sync.log).

Reply:
I notice that in this test the try_until gives up too early. It may be that the initial time of the try_until is miscomputed.

Reply:
I pushed d2a6809 that shall fix this - testing.

Reply:
Passed locally and on CI.